### PR TITLE
Feature/ruff format

### PR DESCRIPTION
**Pull request by zebengberg**
## Internal changes

- Use `ruff` to format the codebase in place of `black`.
- Run `ruff` linting and formatting over the notebook examples in the documentation.
- Update development documentation with new links.

## Tests

- [x] QC test passes locally (`make test`)
- [x] CI tests pass

## Reviewer

> Select @ reviewer

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -71,14 +71,20 @@ dev-pycontrails-bada:
...
+	ruff format pycontrails tests docs
+
+ruff:
+	@echo -e "$(COLOR_YELLOW)The 'ruff' target is deprecated. Use 'make lint' instead.\n"
+	make lint
 
-black-check:
-	black pycontrails tests --check
+black:
+	@echo -e "$(COLOR_YELLOW)The 'black' target is deprecated. Use 'make format' instead.\n"
```
Comment (mlshapiro) [inline Makefile:86]: Is there any reason to keep these? I think we are the only ones with workflows around them

File: pyproject.toml [hunk 2]
```diff
@@ -203,13 +202,6 @@ show_error_codes = true
 # strict_equality = true
 # # --strict end
 
-# black code formatting
-# https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#configuration-via-a-file
-[tool.black]
-line-length = 100
```
Comment (mlshapiro) [inline pyproject.toml:209]: is the default line-length for ruff 100? I don't see this configured below

File: pyproject.toml [hunk 3]
```diff
@@ -203,13 +202,6 @@ show_error_codes = true
...
-preview = true
-required-version = "24.8.0"
```
Comment (mlshapiro) [inline pyproject.toml:211]: is there anything similar for ruff? This is kind of helpful for footgun stuff

**Review by mlshapiro — APPROVED** [review]

File: pyproject.toml (hunk 2 above)
Comment (zebengberg) [reply pyproject.toml:209]: The default is 88, but we'd previously specified this as 100 under `tool.ruff` which is why you don't see a diff there.

File: pyproject.toml (hunk 3 above)
Comment (zebengberg) [reply pyproject.toml:211]: Unfortunately no. We've pinned the ruff version in the dev dependencies, but the ruff configuration doesn't confirm it. As ruff matures more, I think breaking changes will be less common.

**Pull request merged**